### DISCUSSION TITLE
Is it possible to make retry interval unlimited?

### DISCUSSION THREAD

**Discussion by bernard-bear** [emqx/emqx#12945] [General]
Is it possible to change `retry_interval` to an unlimited value, so that there are no retries at all (to be compliant with the MQTT v5 spec)? According to the documentation ([here](https://www.emqx.io/docs/en/latest/design/retransmission.html#basic-configuration) and [here](https://www.emqx.io/docs/en/latest/configuration/mqtt.html#session-settings)), it seems that there are no optional values possible.

Also, when is it scheduled for deprecation? According to #9447, it will be deprecated in v5.

**Comment by ieQu1** [comment]
After briefly looking at the code, I don't think it's possible. Apparently the task of deprecating this parameter fell through the cracks. This should be an relatively easy fix, though.

**Comment by zmstone** [comment]
`retry_interval = "4294967s"` (max timeout config) should make it infinity, effectively.

We'll change it to `infinity` by default in version 5.8. Internal ticket is created to make sure it will not fall through the cracks again.

For EMQX dev team's [internal ref](https://emqx.atlassian.net/browse/EMQX-12281)
